### DISCUSSION TITLE
[Question] How do I bind <C-w>s <C-w>v to an action?

### DISCUSSION THREAD

**Discussion by Ajaymamtora** [ibhagwan/fzf-lua#1487] [Q&A] [answered]
Using this syntax in actions doesnt let me combine 3 keys:
```
      ["ctrl-s"] = actions.file_split, -- <c-w>s
      ["ctrl-v"] = actions.file_vsplit, -- <c-w>v
```

Is this possible?

**Comment by ibhagwan** [comment]
> Using this syntax in actions doesnt let me combine 3 keys:

These are fzf style binds, the key combination you mention in the title isn’t possible, see `man fzf` for supported key combinations.

If you insist on this combo you can still do it albeit differently, use a neovim `vim.keymap.set` in the `winopts.on_create` callback which then uses `nvim_feedkeys` to send keys to the terminal (which you have an fzf mapping for).

**Reply by Ajaymamtora** [reply]
Thanks for the reply, I tried this and nothing happens when I invoke the keybinding:

```lua
{
    defaults = {
      multiline = 2,
    },
    fzf_opts = {
      ["--highlight-line"] = true,
      -- ["--border"] = "none",
      -- ["--layout"] = "reverse",
      -- ["--height"] = "100%",
      -- ["--info"] = "inline-right",
      ["--ansi"] = true,
    },
    fzf_colors = {
      ["fg"] = { "fg", "FzfLuaLine" },
      ["bg"] = { "bg", "FzfLuaNormal" },
      ["hl"] = { "fg", "FzfLuaItemKindVariable" },
      ["fg+"] = { "fg", "FzfLuaLinePlus" },
      ["bg+"] = { "bg", "FzfLuaNormal" },
      ["hl+"] = { "fg", "FzfLuaItemKindVariable" },
      ["info"] = { "fg", "FzfLuaPrompt" },
      ["prompt"] = { "fg", "FzfLuaPrompt" },
      ["pointer"] = { "fg", "DiagnosticError" },
      ["marker"] = { "fg", "DiagnosticError" },
      ["spinner"] = { "fg", "FzfLuaPrompt" },
      ["header"] = { "fg", "FzfLuaPrompt" },
      ["gutter"] = { "bg", "FzfLuaNormal" },
    },
    winopts = {
      -- ["none"]    = { "", "", "", "", "", "", "", "" },
      -- ["empty"]   = { " ", " ", " ", " ", " ", " ", " ", " " },
      -- ["single"]  = { "┌", "─", "┐", "│", "┘", "─", "└", "│" },
      -- ["double"]  = { "╔", "═", "╗", "║", "╝", "═", "╚", "║" },
      -- ["rounded"] = { "╭", "─", "╮", "│", "╯", "─", "╰", "│" },
      -- ["thicc"]   = { "┏", "━", "┓", "┃", "┛", "━", "┗", "┃" },
      -- ["thiccc"]  = { "▛", "▀", "▜", "▐", "▟", "▄", "▙", "▌" },
      -- ["thicccc"] = { "█", "█", "█", "█", "█", "█", "█", "█" },
      -- win_border = { "╭", "─", "╮", "│", "╯", "─", "╰", "│" },
      win_border = { "┌", "─", "┐", "│", "┘", "─", "└", "│" },
      hl_normal = "Normal", -- window normal color
      hl_border = "FloatBorder", -- window border color
      title = "FZF",
      title_pos = "center",
      width = 0.9,
      height = 0.9,
      preview = {
        layout = "horizontal",
      },
      on_create = function()
        vim.keymap.set("i", "<M-c>", function()
          print("AAAA")
          return require("fzf-lua").actions.resume()
        end, { nowait = true, buffer = true })

        vim.keymap.set("i", "<M-l>", function()
          print("AAAA")
          return require("fzf-lua").actions.resume()
        end, { nowait = true, buffer = true })
      end,
    },
    grep = {
      prompt = " ",
      rg_opts = "--column --line-number --glob='!.git/' --no-heading --color=always --no-ignore --hidden --fixed-strings --smart-case --max-columns=4096 -e",
      no_header_i = true,
      rg_glob = true,
      exec_empty_query = true,
    },
  }
```

am I missing something?

**Reply by ibhagwan** [reply]
I’m honestly not sure what you’re trying to do here, if all you need is Alt bindings those can be used directly in keymap.fzf as `alt-l` or `alt-c`, what’s the point of calling `resume` while the interface open? You’re just triggering a process reset for no reason.

**Reply by Ajaymamtora** [reply]
I want a uniform way of toggling things, so I don't have to worry about cross platform differences which nvim bindings allow me to do. I don't know if the fzf bindings allow this.

Ignore whats going on inside the keymaps, the point is they do not do anything.

It would be useful if a buffer number was provided in the on_create callback so i could use that in the mapping, just incase thats the issue here.

I'm trying to migrate from telescope, and another thing I had setup was to open grug-far with the current file mask, literal etc from the current fzf picker. I thought doing all this would make sense in the on_create instead of the fzf style bindings. I don't care that the process resets.

**Comment by ibhagwan** [comment]
> It would be useful if a buffer number was provided in the on_create callback so i could use that in the mapping, just incase thats the issue here.

I can add that easily if necessary, for now you can use `vim.api.nvim_get_current_buf()`.

**Comment by Ajaymamtora** [comment, marked as answer]
Just needed to set the mapping for terminal mode
